Makefile

```diff
@@ -1,92 +1,141 @@
-.PHONY: all binary test plugin ipam ut clean update-version
+SRCFILES=$(wildcard *.go) $(wildcard utils/*.go) $(wildcard test_utils/*.go) ipam/calico-ipam.go
+LOCAL_IP_ENV?=$(shell ip route get 8.8.8.8 | head -1 | cut -d' ' -f8)
 
-# Version of calico/build to use.
-BUILD_VERSION=latest
+K8S_VERSION=1.2.4
+CALICO_NODE_VERSION=0.19.0
 
-SRCFILES=$(shell find calico_cni -type f ! -path calico_cni/version.py) calico.py ipam.py
-LOCAL_IP_ENV?=$(shell ip route get 8.8.8.8 | head -1 | cut -d' ' -f8)
+# Ensure that the dist directory is always created
+MAKE_SURE_DIST_EXIST := $(shell mkdir -p dist)
 
-K8S_VERSION=1.2.0
+BUILD_CONTAINER_NAME=calico/cni_container
+BUILD_CONTAINER_MARKER=cni_container.created
 
+.PHONY: all binary plugin ipam
 default: all
-all: binary test
-binary: update-version dist/calico dist/calico-ipam
-test: ut fv
+all: vendor build-containerized test-containerized
+binary:  plugin ipam
 plugin: dist/calico
 ipam: dist/calico-ipam
 
+.PHONY: clean
+clean:
+	-sudo rm -rf dist vendor
 
-# Builds the Calico CNI plugin binary.
-dist/calico: $(SRCFILES) 
-	docker run  --rm \
-	-v `pwd`:/code \
-	calico/build:$(BUILD_VERSION) \
-	/bin/sh -c "pip install pykube && pyinstaller calico.py -ayF"
+# Updates all the vendored libraries, i.e. changes the contents of the glide.lock. Running this should NOT be a common operation.
```

Comment:
Since this isn't a common operation, it should probably include a comment explaining when it should be run.